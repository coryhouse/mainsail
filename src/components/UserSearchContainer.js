import React, {Component} from 'react';
import UserSearch from "./UserSearch";
import { connect } from "react-redux";
import * as actions from "../actions/userSearchActions";
import { bindActionCreators } from "redux";

class UserSearchContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstName: '',
      lastName: ''
    };
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value});
  };

  handleSearchRequest = (event) => {
    event.preventDefault();
    this.props.actions.searchUsers({
      firstName: this.state.firstName,
      lastName: this.state.lastName
    });
  };

  render() {
    const {firstName, lastName} = this.state;
    return <UserSearch onSearch={this.handleSearchRequest} onChange={this.handleChange} firstName={firstName} lastName={lastName} />
  }
}

// Connect specific Redux store data to component above
function mapStateToProps(state) {
  return {
    users: state.users
  };
}

// Connect specific Redux actions to component above
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(UserSearchContainer);