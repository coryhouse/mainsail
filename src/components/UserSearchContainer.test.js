import React from "react";
import { UserSearchContainer } from "./UserSearchContainer";
import renderer from "react-test-renderer";


// it('should hide userList when users do not exist', () => {
//   const tree = renderer
//     .create(<UserSearchContainer users={[]} actions={{}} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

it('should hide userList when users do not exist', () => {
  const users = [
    {
      firstName: "Mark",
      lastName: "Longman",
      userID: "MLONGMAN"
    }
  ];

  const tree = renderer
    .create(<UserSearchContainer users={users} actions={{searchUsers:() => {}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});