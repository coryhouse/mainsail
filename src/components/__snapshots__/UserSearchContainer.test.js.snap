// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide userList when users do not exist 1`] = `
<div>
  <form>
    <label>
      First Name
    </label>
    <input
      className="uppercase"
      name="firstName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Last Name
    </label>
    <input
      name="lastName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      onClick={[Function]}
      type="submit"
      value="Search"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </form>
  <div>
    <h2>
      1
       Search Results
    </h2>
    <table
      className="ui small celled selectable sortable table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            #
          </th>
          <th
            className=""
          >
            Name
          </th>
          <th
            className=""
          >
            Email
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      >
        <tr
          className=""
        >
          <td
            className=""
          >
            1
            .
          </td>
          <td
            className=""
          >
            Mark Longman
          </td>
          <td
            className=""
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
